bbc/simorgh
# `getEmbedUrl` refactor + Metadata component for av-embeds

Overall changes
======
- Refactors the `getEmbedUrl` function and where its called within the new MediaLoader component
- This ensures parity with the existing use of this function for setting the Morph embed URLs and should cater for both CPS and Optimo URL structures
- Adds new `Metadata` component to MediaLoader to set the embedUrl and JSON+LD values for SEO

Code changes
======
- Adds calls to `getEmbedUrl` within each SMP config builder, as the values needed to construct the URL differ between configs
- Removes call to `getEmbedUrl` from `handleAvRoute`, instead relying on the new `Metadata` component to set the `og:url` value there

Testing
======
1. Visit http://localhost:7081/ws/av-embeds/articles/cd1rmn075d1o/p0jd37n8/ig?renderer_env=live
2. Right click on the page and open 'View Page Source'
3. Search for `og:url` and confirm its set to `https://www.test.bbc.com/ws/av-embeds/articles/cd1rmn075d1o/p0jd37n8/ig`
4. Search for `ld+json` and confirm that a JSON object has been set within the `script` tag
5. Visit http://localhost:7081/serbian/cyr/av-embeds/srbija-68707945?renderer_env=live
6. Repeat and confirm the steps above for this CPS asset

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
